### DISCUSSION TITLE
Tweak Addition: In-Game Shortcut Modifier for Retroarch (or other systems)

### DISCUSSION THREAD

**Discussion by braedon2011** [OnionUI/Onion#965] [Ideas]
It would be great to have a feature like this that is easily accessible to the user instead of switching or disabling commands in Retroarch. Too many of my friends who I have made devices for have accidentally loaded a previous save state and lost all their progress when hitting the wrong trigger while rewinding or fast-forwarding.

**Comment by J1C0** [comment]
Here's a possible solution:
When Retroarch would perform a Load State action, instead have it do this:
- Increment save slot +1
- Save state
- Increment save slot -1
- Load State

99% of the time you'll never need the save that this macro makes, and you'll automatically save over it the next time you save your state.
But the Save State just prior to your loading action is available right after you load, just in case you need it.
